service/controller/resource/promxy/servergroup.go

```diff
@@ -0,0 +1,172 @@
+package promxy
+
+import (
+	"fmt"
+	"time"
+
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/pkg/relabel"
+)
+
+// Config is the configuration for a ServerGroup that promxy will talk to.
+// This is where the vast majority of options exist.
+type ServerGroup struct {
```

Comment:
ServerGroup implementation should be replace with https://godoc.org/github.com/jacksontj/promxy/pkg/servergroup#Config

Or please explain what changed here, otherwise it's hard to track and see the difference.

Reply:
Sure, what changed is that promxy is using a fork of prometheus 1.8 as a dependency while we rely on 2.20 and a lot of things changed since then so I had to do a few changes to the ServerGroup struct like: // Hosts is a set of discovery.Config options that allow promxy to discover
	// all hosts in the server_group
	KubernetesSDConfigs []*kubernetes.SDConfig `yaml:"kubernetes_sd_configs,omitempty"`